src/test/ui/exclusive-range/exclusive_range_pattern_syntax_collision.rs

```diff
@@ -2,7 +2,10 @@
 
 fn main() {
     match [5..4, 99..105, 43..44] {
-        [_, 99.., _] => {}, //~ ERROR unexpected token: `,`
+        [_, 99.., _] => {},
+        //~^ ERROR `X..` range patterns are not supported
+        //~| ERROR arbitrary expressions aren't allowed in patterns
+        //~| ERROR only char and numeric types are allowed in range patterns
```

Comment:
Both these errors could use an exception for `ExprKind::Err`/`TyKind::Error`.
(Error nodes/types are supposed to suppress further diagnostics in general.)